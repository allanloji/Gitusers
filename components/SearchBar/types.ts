type SearchBarProps = {};

export type { SearchBarProps };
