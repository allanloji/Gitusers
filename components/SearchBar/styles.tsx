import styled from '@emotion/styled';

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: white;
  height: 50px;
  width: 90%;
  border-radius: 50px;
  padding: 0 1.5rem;
`;

const TextInput = styled.input`
  border: none;
  width: 100%;
  font-size: 20px;
  :focus {
    outline: none !important;
  }
`;

const SearchButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
`;

const SearchIcon = styled.img`
  width: 30px;
  height: 30px;
`;

export { Container, TextInput, SearchButton, SearchIcon };
