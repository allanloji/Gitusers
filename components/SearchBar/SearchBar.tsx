import React, { useState } from 'react';

import { Container, SearchButton, SearchIcon, TextInput } from './styles';
import { SearchBarProps } from './types';

/**
 * SearchBar
 */
function SearchBar({ onSubmit }: SearchBarProps) {
  const [search, setSearch] = useState('');

  return (
    <Container>
      <TextInput
        placeholder='Search by name'
        value={search}
        onChange={event => setSearch(event.target.value)}
      />
      <SearchButton
        data-testid='submit-bttn'
        onClick={() => {
          if (search !== '') {
            onSubmit(search);
          }
        }}
      >
        <SearchIcon src='/assets/search.svg' alt='Search icon' />
      </SearchButton>
    </Container>
  );
}

export default SearchBar;
