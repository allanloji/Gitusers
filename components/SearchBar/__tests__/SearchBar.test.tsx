import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SearchBar from '..';

describe('SearchBar', () => {
  test('Should render', () => {
    const onSubmit = jest.fn();

    const { container } = render(<SearchBar onSubmit={onSubmit} />);

    expect(container).toMatchSnapshot();
  });

  test('Should send text written', () => {
    const onSubmit = jest.fn();

    const { getByPlaceholderText, getByTestId } = render(<SearchBar onSubmit={onSubmit} />);

    const input = getByPlaceholderText('Search by name');
    const submit = getByTestId('submit-bttn');

    userEvent.type(input, 'Test');
    userEvent.click(submit);

    expect(onSubmit).toHaveBeenCalledWith('Test');
  });

  test('Should send not send empty text', () => {
    const onSubmit = jest.fn();

    const { getByTestId } = render(<SearchBar onSubmit={onSubmit} />);

    const submit = getByTestId('submit-bttn');

    userEvent.click(submit);

    expect(onSubmit).not.toHaveBeenCalled();
  });
});
