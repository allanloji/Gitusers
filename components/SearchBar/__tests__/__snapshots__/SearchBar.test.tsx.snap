// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should render 1`] = `
<div>
  <div
    class="css-1wisk8h"
  >
    <input
      class="css-1bfqk3l"
      placeholder="Search by name"
      value=""
    />
    <button
      class="css-zq0wcq"
      data-testid="submit-bttn"
    >
      <img
        alt="Search icon"
        class="css-ulohxf"
        src="/assets/search.svg"
      />
    </button>
  </div>
</div>
`;
